src/Solvers/Greedy.jl

```diff
@@ -98,7 +97,7 @@ function ssa_greedy_optimize(inputs, output, size, choose_fn=greedy_choose_simpl
     while !isempty(queue)
         # select candidate
         winner = choose_fn(queue, remaining)
-        if winner == nothing
+        if winner === nothing
```

Comment:
Is there any difference with `winner == nothing`?

This is ok. I'm just curious.

Reply:
The `===` are an alias for the `is` function, which checks if two objects are programmatically indistinguishable. I have read somewhere that it should be more efficient.

Reply:
Well, that's not completely correct.

The `isa` (not `is`) function checks whether an object `x` is of given type (e.g. `x isa Int` if `x` is of type `Int`).
The `===` function checks whether 2 objects are _identical_ (i.e. they are the same object). This is different from `==` because 2 objects can have the same type and the same content but can be different objects. For example,
```julia
julia> a = [1,2]

julia> b = [1,2]

julia> a == b
true

julia> a === b
false
```

Summarizing, `===` implies `==` but not the other way around.

Reply:
In this case, the code is correct because `nothing` is of type `Nothing` which is a **singleton** type (i.e. a type that can only have a single instance/object posible) and thus any instance of `Nothing` are the same object in reality (i.e. only one `nothing` object exists).